src/main/java/command/DoneCommand.java

```diff
@@ -0,0 +1,40 @@
+package command;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+import static java.lang.String.format;
+
+/**
+ * Marks a task as done.
+ *
+ */
+public class DoneCommand extends Command {
+    private int index;
+
+    /**
+     * Marks a task as done.
+     *
+     * @param input Given user input.
+     * @throws DukeException If any error is encountered.
+     */
+    public DoneCommand(String input) throws DukeException {
+        this.index = Integer.parseInt(input);
```

Comment:
`NumberFormatException` should be caught here as well I think (similar to what you did in `DeleteCommand`)?